src/components/EpisodePreview/EpisodePreview.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useContext } from "react";
+import {
+	EpisodesContext as Episodes
+} from "COMPONENTS/Episodes";
+
+export default function EpisodePreview(props){
+
+	// CONTEXT
+	//-----------------------------
+	const episodes = useContext(Episodes);
+
+
+	// RENDER
+	// ----------------------------
+	const { 
+		episodeId,
+		index,
+		episodeNo,
+		title,
+		thumbnail: {
+			src: thumbSrc,
+			description: thumbDesc
+		},
+		trailer: {
+			src: trailerSrc
+		}
+	} = props;
+
+	// handle visibility of active episode and episode preview overlay
+	const isActiveEpisode 	= episodes.get.activeIndex === index;
+	const thisOverlayActive = isActiveEpisode && episodes.get.overlayActive;
+	const hidden 			= !thisOverlayActive;
+
+	const isFirst 	= index === 0;
+	const isLast 	= index === episodes.get.episodeCount - 1;
+
+	const prevEpisodeNo = episodeNo - 1;
+	const nextEpisodeNo = episodeNo + 1;
+
+	const previewId = title
+		.replace(/\W+/g, '-') // remove non-alphanumeric & replace with -s
+		.toLowerCase()
+		+"-preview"
+		;
+
+	const prevPreviewId = `episode__${prevEpisodeNo}__preview`;
+	const nextPreviewId = `episode__${nextEpisodeNo}__preview`;
+
+	function handleNavPressed(e, newIndex){
```

Comment:
This should be `setActiveIndex` or something - `handleSomething` is used when the function is doing so many things that you can't really give it a useful name and so you're just like IT'S THE THING THAT HAPPENS WHEN YOU CLICK IT.

Reply:
don't forget about this!